CanvasColorSpaceProposal.md

```diff
@@ -164,43 +159,67 @@ Note: An alternative approach that was considered was to augment the probablySup
 
 #### ImageBitmap
 
-TODO(ccameron): Review this
-ImageBitmap objects are augmented to have an internal color space attribute of type CanvasColorSpace and an internal pixelFormat attribute of type CanvasPixelFormat. The colorSpaceConversion creation attribute also accepts enum values that correspond to CanvasColorSpace values. Specifying a CanvasColorSpace value results in a conversion of the image to the specified color space.
+ImageBitmap objects (unless created with ``colorSpaceConversion="none"``) should keep track of their internal color space, and should store their contents at highest fidelity possible, subject to implementation limitations.
 
 #### ImageData
 
-TODO(ccameron): Review this
-
-IDL
+Add the following types to be used by `ImageData`.
 <pre>
-
 enum ImageDataStorageType {
   "uint8", // default
   "uint16",
   "float32",
 };
-
+dictionary ImageDataSettings {
+  CanvasColorSpaceEnum colorSpace = "srgb";
+  ImageDataStorageType storageType = "uint8";
+};
 typedef (Uint8ClampedArray or Uint16Array or Float32Array) ImageDataArray;
+</pre>
 
-[Constructor(unsigned long sw, unsigned long sh, optional ImageDataColorSettings imageDataColorSettings),
- Constructor(ImageDataArray data, unsigned long sw, optional unsigned long sh, optional ImageDataColorSettings imageDataColorSettings),
- Exposed=(Window,Worker)]
-interface ImageData {
-  readonly attribute unsigned long width;
-  readonly attribute unsigned long height;
+Update the `ImageData` interface to the include the following.
+<pre>
+partial interface ImageData {
+  constructor(unsigned long sw, unsigned long sh, optional ImageDataSettings);
+  constructor(ImageDataArray data, unsigned long sw, unsigned long sh, optional ImageDataSettings);
+  readonly ImageDataSettings getImageDataSettings();
   readonly attribute ImageDataArray data;
-
-  ImageDataColorSettings getColorSettings();
 };
 </pre>
 
-* When using the constructor that takes an ImageDataArray parameter, the "storageType" setting is ignored.
-* createImageData() and getImageData() produce an ImageData object with the same color space as the source canvas, using an ImageDataArray of a type that is appropriate for the pixelFormat of the source canvas (smallest possible numeric size that guarantees no loss of precision).
-* putImageData() performs a color space conversion to the color space of the destination canvas.
-* Data returned by getImageData() or passed to putImageData() are assumed to be in "srgb" color space, with "unorm8" encoding.
+The changes to this interface are:
+* The constructors now take an optional `ImageDataSettings` dictionary.
+* The ImageDataSettings attribute may be queried using `getImageDataSettings`.
```

Comment:
I guess the only remaining question is whether this should be a readonly attribute or a method. Right now the interface contains only attributes.

Reply:
I'll have to defer to people who are more expert than me (I don't have enough experience in these APIs to have a sense as to what is natural)

Reply:
I'll go ahead and merge this, but we should continue to revise it.